common/src/main/java/com/wynntils/utils/mc/ComponentUtils.java

```diff
@@ -168,4 +176,42 @@ public static MutableComponent makeObfuscated(
         }
         return newName;
     }
+
+    /**
+     * Adds a Wynntils chat banner to the left side of the provided text in the style of Wynncraft 2.1 chat banners.
+     * The formatting of the provided text is preserved.
+     *
+     * @param formattedText the formatted text to add the Wynntils chat banner to
+     * @return a {@code Component} holding the formatted text with the Wynntils chat banner added
+     */
+    public static Component addWynntilsBanner(FormattedText formattedText) {
```

Comment:
I don't get it..? This takes a FormattedText, not a Component? But it's called with a Component? How does this even compile?

Reply:
Ah, right, FormattedText, not StyledText. Nevertheless, maybe it should be Component anyway, for consistency?

Reply:
It was originally component but the splitcomponent method didn’t work right. It split each character instead of splitting by my width, so I used another way of splitting. I’ll see if I can make it work with component.